applications/StructuralMechanicsApplication/custom_strategies/custom_schemes/explicit_central_differences_scheme.hpp

```diff
@@ -512,7 +506,7 @@ namespace Kratos
             
         }//for DoF
         ////// ROTATION DEGRESS OF FREEDOM
-        if (i_begin->HasDofFor(ROTATION_X))
```

Comment:
i am suggesting you to do this check out of the node loop. If the first node has the dof ROTATION_Z all the others in the modelpart will also have ti. This will have a sensible effect on performance

Reply:
to make it clear, the performance problem is the function HasDofFor, which internally does a search.

so i am telling to define outside the loop a local variable

    const has_dof_for_rotation_z = model_part.NodesBegin()->HasDofFor(ROTATION_Z)

and use this variable instead